cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/service/MetadataStore.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.metadata.service;
+
+import co.cask.cdap.data2.metadata.dataset.MetadataDataset;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.metadata.MetadataRecord;
+import co.cask.cdap.proto.metadata.MetadataScope;
+import co.cask.cdap.proto.metadata.MetadataSearchResultRecord;
+import co.cask.cdap.proto.metadata.MetadataSearchTargetType;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Defines operations on {@link MetadataDataset} for both system and user metadata.
+ *
+ * Operations supported for a specified {@link Id.NamespacedId}:
+ * <ul>
+ *   <li>Adding new metadata: Supported for a single scope.</li>
+ *   <li>Retrieving metadata: Supported for a specified scope as well as across both scopes.</li>
+ *   <li>Removing metadata: Supported for a specified scope as well as across both scopes.</li>
+ * </ul>
+ */
+public interface MetadataStore {
```

Comment:
Lets says in function docs which does not have scopes that it will be scope for clarity.